email/email.go

```diff
@@ -55,10 +60,16 @@ type emailAgent struct {
 	webRedirectDir webRedirectDirConfig
 }
 
-func (this *emailAgent) WebRedirectDir(success bool) string {
+func (this *emailAgent) WebRedirectDir(success bool, authType string) string {
 	if success {
+		if authType == ReauthorizeType {
+			return this.webRedirectDir.WebRedirectDirOnReauthorizeSuccess
+		}
 		return this.webRedirectDir.WebRedirectDirOnSuccess
 	}
+	if authType == ReauthorizeType {
+		return this.webRedirectDir.WebRedirectDirOnReauthorizeFailure
+	}
```

Comment:
func (this *emailAgent) WebRedirectDir(success bool, authType string) string {
    return this.webRedirectDirConfigs.get(success, authType )
}